src/main/java/com/iexec/worker/sgx/SgxService.java

```diff
@@ -44,21 +56,42 @@ public SgxService(
     ) {
         this.workerConfigService = workerConfigService;
         this.dockerService = dockerService;
-        this.sgxEnabled = !SgxDriverMode.NONE.equals(sgxDriverMode) && isSgxSupported();
+        this.sgxDriverMode = sgxDriverMode;
+    }
+
+    @PostConstruct
+    void init() {
+        boolean sgxSupported = isSgxSupported(sgxDriverMode);
+        final boolean driverModeNotNone = SgxDriverMode.isDriverModeNotNone(sgxDriverMode);
+        if (!sgxSupported && driverModeNotNone) {
+            this.sgxEnabled = false;
+            log.error("SGX required but not supported by worker. Shutting down. " +
+                    "[sgxDriverMode: {}]", sgxDriverMode);
+            SpringApplication.exit(context, () -> 1);
+            return;
+        }
+
+        this.sgxEnabled = driverModeNotNone;
+    }
+
+    public SgxDriverMode getSgxDriverMode() {
+        return sgxDriverMode;
     }
 
     public boolean isSgxEnabled() {
         return sgxEnabled;
     }
 
-    private boolean isSgxSupported() {
+    private boolean isSgxSupported(SgxDriverMode sgxDriverMode) {
         log.info("Checking SGX support");
-        boolean isSgxDriverFound = new File(SgxUtils.SGX_DRIVER_PATH).exists();
-        if (!isSgxDriverFound) {
-            log.error("SGX driver not found");
-            return false;
+        if (sgxDriverMode == SgxDriverMode.LEGACY) {
+            boolean isSgxDriverFound = new File(SgxUtils.SGX_DRIVER_PATH).exists();
```

Comment:
```suggestion
            boolean isLegacySgxDriverFound = new File(SgxUtils.LEGACY_SGX_DRIVER_PATH).exists();
```
With the needed modifications of course.

Reply:
Done in https://github.com/iExecBlockchainComputing/iexec-worker/pull/434/commits/4fb0cb27c405a723adfe394e21374fd790f66c04.